src/permissions.rs

```diff
@@ -36,60 +38,80 @@ impl DenoPermissions {
       return Ok(());
     };
     // TODO get location (where access occurred)
-    let r = permission_prompt("access to run a subprocess");
-    if r.is_ok() {
-      self.allow_run.store(true, Ordering::SeqCst);
+    if !self.no_prompts.load(Ordering::SeqCst) {
+      let r = permission_prompt("access to run a subprocess");
+      if r.is_ok() {
+        self.allow_run.store(true, Ordering::SeqCst);
+      }
+      r
+    } else {
+      Err(permission_denied())
     }
-    r
   }
 
   pub fn check_read(&self, filename: &str) -> DenoResult<()> {
     if self.allow_read.load(Ordering::SeqCst) {
       return Ok(());
     };
     // TODO get location (where access occurred)
-    let r = permission_prompt(&format!("read access to \"{}\"", filename));;
-    if r.is_ok() {
-      self.allow_read.store(true, Ordering::SeqCst);
+    if !self.no_prompts.load(Ordering::SeqCst) {
+      let r = permission_prompt(&format!("read access to \"{}\"", filename));;
+      if r.is_ok() {
+        self.allow_read.store(true, Ordering::SeqCst);
+      }
+      r
+    } else {
+      Err(permission_denied())
     }
-    r
   }
 
   pub fn check_write(&self, filename: &str) -> DenoResult<()> {
     if self.allow_write.load(Ordering::SeqCst) {
       return Ok(());
     };
     // TODO get location (where access occurred)
-    let r = permission_prompt(&format!("write access to \"{}\"", filename));;
-    if r.is_ok() {
-      self.allow_write.store(true, Ordering::SeqCst);
+    if !self.no_prompts.load(Ordering::SeqCst) {
+      let r = permission_prompt(&format!("write access to \"{}\"", filename));;
+      if r.is_ok() {
+        self.allow_write.store(true, Ordering::SeqCst);
+      }
+      r
+    } else {
+      Err(permission_denied())
     }
-    r
   }
 
   pub fn check_net(&self, domain_name: &str) -> DenoResult<()> {
     if self.allow_net.load(Ordering::SeqCst) {
       return Ok(());
     };
     // TODO get location (where access occurred)
-    let r =
-      permission_prompt(&format!("network access to \"{}\"", domain_name));
-    if r.is_ok() {
-      self.allow_net.store(true, Ordering::SeqCst);
+    if !self.no_prompts.load(Ordering::SeqCst) {
+      let r =
+        permission_prompt(&format!("network access to \"{}\"", domain_name));
+      if r.is_ok() {
+        self.allow_net.store(true, Ordering::SeqCst);
+      }
+      r
+    } else {
+      Err(permission_denied())
```

Comment:
Seems like there are 4 very similar sections of code here. Is it possible to "DRY" this up?

Reply:
That should be much cleaner.